packages/insomnia-app/app/ui/containers/app.js

```diff
@@ -1027,37 +1026,43 @@ class App extends PureComponent {
     }
 
     if (activeGitRepository) {
-      // Create FS plugin
+      // Create FS client
       const baseDir = path.join(
         getDataDirectory(),
         `version-control/git/${activeGitRepository._id}`,
       );
-      const pNeDb = NeDBPlugin.createPlugin(activeWorkspace._id);
-      const pGitData = FSPlugin.createPlugin(baseDir);
-      const pOtherData = FSPlugin.createPlugin(path.join(baseDir, 'other'));
-
-      const fsPlugin = routableFSPlugin(
-        // All data outside the directories listed below will be stored in an 'other'
-        // directory. This is so we can support files that exist outside the ones
-        // the app is specifically in charge of.
-        pOtherData,
-        {
-          // All app data is stored within the a namespaced directory at the root of the
-          // repository and is read/written from the local NeDB database
-          [GIT_INSOMNIA_DIR]: pNeDb,
-
-          // All git metadata is stored in a git/ directory on the filesystem
-          [GIT_INTERNAL_DIR]: pGitData,
-        },
-      );
+
+      /** All app data is stored within the a namespaced directory at the root of the repository and is read/written from the local NeDB database */
+      const neDbClient = NeDBClient.createClient(activeWorkspace._id);
+
+      /** All git metadata is stored in a git/ directory on the filesystem */
+      const gitDataClient = fsClient(baseDir);
+
+      /** All data outside the directories listed below will be stored in an 'other' directory. This is so we can support files that exist outside the ones the app is specifically in charge of. */
+      const otherDatClient = fsClient(path.join(baseDir, 'other'));
+
+      const routableFS = routableFSClient(otherDatClient, {
```

Comment:
A comment to explain how the above clients (and the purpose they serve) are orchestrated by the `routableFSClient` would be useful. Something like:

```suggestion
      /** The routable FS client directs isomorphic-git to read/write from the database or from the correct directory on the file system while performing git operations. */
      const routableFS = routableFSClient(otherDatClient, {
```